compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1190,7 +1190,8 @@ object RefChecks {
             }
           }
         .exists
-      if !target.typeSymbol.denot.isAliasType && !target.typeSymbol.denot.isOpaqueAlias && hidden
+      if !target.typeSymbol.denot.isOpaqueAlias // Opaque types hide original members so extension is not nullified by the original member
```

Comment:
nit: I don't think the comment is necessary, it is one of the main two features of `opaque` types.
Also, you can get rid of the `.denot`, since it's implied.

Reply:
I sympathize but putting the conditional on 2 lines makes it harder to read. If the comment is necessary, I'd put it on its own line. If there were several complex subexpressions, comment per line would be reasonable.

Reply:
Thanks! Removed the comment and put the conditional on 1 line